core/teca_algorithm.cxx

```diff
@@ -121,14 +127,50 @@ teca_algorithm_internals::teca_algorithm_internals()
     name("teca_algorithm"),
     data_cache_size(1),
     modified(1),
-    exec(teca_algorithm_executive::New())
+    exec(teca_algorithm_executive::New()),
+#if defined(TECA_HAS_MPI)
```

Comment:
@burlen - could you please clarify what this block is doing (perhaps clarifying w/ comments in the code)?

Is this setting a default for the communicator?

Reply:
yes, that is where the default communicator is set. If no one calls `set_communicator` then then `MPI_COMM_WORLD` is returned from `get_communicator`. I'll add a comment.